pkgs/development/coq-modules/mathcomp-analysis/default.nix

```diff
@@ -36,20 +37,24 @@ let
   packages = [ "classical" "analysis" ];
 
   mathcomp_ = package: let
-      analysis-deps = map mathcomp_ (head (splitList (pred.equal package) packages));
-      pkgpath = if package == "analysis" then "theories" else "${package}";
-      pname = "mathcomp-${package}";
+      classical-deps = [ mathcomp.algebra mathcomp-finmap hierarchy-builder ];
+      analysis-deps = [ mathcomp.field mathcomp-bigenough ];
+      intra-deps = if package == "single" then []
+        else map mathcomp_ (head (splitList (pred.equal package) packages));
+      pkgpath = if package == "single" then "."
+        else if package == "analysis" then "theories" else "${package}";
+      pname = if package == "single" then "mathcomp-analysis-single"
+        else "mathcomp-${package}";
       derivation = mkCoqDerivation ({
         inherit version pname defaultVersion release repo owner;
 
         namePrefix = [ "coq" "mathcomp" ];
 
         propagatedBuildInputs =
-          (if package == "classical" then
-             [ mathcomp.ssreflect mathcomp.algebra mathcomp-finmap ]
-           else
-             [ mathcomp.field mathcomp-bigenough mathcomp-real-closed ])
-          ++ [ analysis-deps ];
+          if package == "single" then classical-deps ++ analysis-deps else
+            (if package == "classical" then classical-deps
+             else analysis-deps)
+            ++ [ intra-deps ];
```

Comment:
```suggestion
          intra-deps
          ++ optionals (elem package [ "classical" "single" ]) classical-deps
          ++ optionals (elem package [ "analysis" "single" ]) analysis-deps
```

Reply:
@proux01 I suggest this refactoring

Reply:
@CohenCyril thanks, done, CI green again: https://github.com/coq-community/coq-nix-toolbox/pull/122